runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -597,10 +607,58 @@ Table<K, V, M> openTable(@Nonnull final String namespace,
                 mapSupplier);
         tableMap.put(fullyQualifiedTableName, (Table<Message, Message, Message>) table);
 
+        resetIfTableTrimmed(namespace, tableName);
         registerTable(namespace, tableName, kClass, vClass, mClass, tableOptions);
         return table;
     }
 
+    /**
```

Comment:
I would also add that this 'reset' is only effective before the next compaction cycle, when the reset itself gets trimmed.